client/src/app/services/profile.service.ts

```diff
@@ -0,0 +1,39 @@
+import { Injectable } from '@angular/core';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class ProfileService {
+
+  constructor() { }
+
+  publications: Array<any> = [];
```

Comment:
You can now make this an array of "Post" (instead of any, Post[]), as the interface was included with the details page pr